### PR TITLE
bug/ added delay publishing event when entity is in context manager

### PR DESCRIPTION
**Pull request by toan-quach**
The property class still posts some pain when updating it in context manager. I tried many different ways but the problem lies down to the object references and the fact that we're storing `_is_in_context` (the attribute used to determine the existence of a context manager) in the memory. And as we have multiple places where we would reload the entity, this `_is_in_context` can be affected. I have checked multiple location and it seems that we're good (fingers-crossed), aside for places like properties.

### CODE REVIEW

**Review by jrobinAV — CHANGES_REQUESTED** [review]
I think we can merge this part and create a new PR for the TODOS. 
However, I prefer to block it until the 2.3.0 is out.

**Comment by toan-quach** [comment]
@jrobinAV Agree! 😃

File: src/taipy/core/_entity/_properties.py [hunk 1]
```diff
@@ -18,19 +18,24 @@ class _Properties(UserDict):
     def __init__(self, entity_owner, **kwargs):
         super().__init__(**kwargs)
         self._entity_owner = entity_owner
+        self._current_in_context_entity_owner = None
```
Comment (jrobinAV) [inline src/taipy/core/_entity/_properties.py]: Should it be part of the other PR?

File: src/taipy/core/cycle/cycle.py [hunk 2]
```diff
@@ -132,6 +132,7 @@ def name(self, val):
     @property
     def properties(self):
         self._properties = _reload(self._MANAGER_NAME, self)._properties
+        self._properties._entity_owner = self
```
Comment (jrobinAV) [inline src/taipy/core/cycle/cycle.py]: I don't like this. It looks like a code smell to me.

File: src/taipy/core/_entity/_properties.py (hunk 1 above)
Comment (toan-quach) [reply src/taipy/core/_entity/_properties.py]: oh my bad yes!!

File: src/taipy/core/cycle/cycle.py (hunk 2 above)
Comment (toan-quach) [reply src/taipy/core/cycle/cycle.py]: I won't argue with that 😅 but other than that, I'm not sure how to solve object references problem

Comment (trgiangdo) [reply src/taipy/core/cycle/cycle.py]: From what I saw, all the
```python
@property
    def properties(self):
```
are the same for all entities.
I think it makes sense to move it into the _Entity class.

And about the `_entity_owner` reference, maybe it is cleaner if we can do something inside the `_reload()` method. But I'm not sure if it is possible or not.

**Review by jrobinAV — APPROVED** [review]

**Review by jrobinAV — APPROVED** [review]
Sounds good to me

**Review by trgiangdo — APPROVED** [review]

**Pull request merged**